packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/editingcontext/migration/participants/NodeDescriptionLayoutStrategyMigrationParticipant.java

```diff
@@ -0,0 +1,113 @@
+import org.eclipse.sirius.emfjson.resource.JsonResource;
+import org.springframework.stereotype.Service;
+
+/**
+ * MigrationParticipant that move the childrenLayoutStrategy from NodeDescription to NodeStyleDescription.
+ *
+ * @author frouene
+ */
+@Service
+public class NodeDescriptionLayoutStrategyMigrationParticipant implements IMigrationParticipant {
+
+    private static final String PARTICIPANT_VERSION = "2025.6.0-202505281000";
+    private static final String CHILDREN_LAYOUT_STRATEGY = "childrenLayoutStrategy";
+    private static final String DATA = "data";
+    private static final String STYLE = "style";
+    private static final String CONDITIONAL_STYLES = "conditionalStyles";
+    private static final String CHILDREN_DESCRIPTIONS = "childrenDescriptions";
+
+    @Override
+    public String getVersion() {
+        return PARTICIPANT_VERSION;
+    }
+
+    @Override
+    public void preDeserialization(JsonResource resource, JsonObject jsonObject) {
+        this.moveLayoutStrategies(jsonObject);
+    }
+
+    public void moveLayoutStrategies(JsonObject jsonObject) {
+        for (String key : jsonObject.keySet()) {
+            JsonElement value = jsonObject.get(key);
+
+            if (value.isJsonObject()) {
+                if (key.equals(CHILDREN_LAYOUT_STRATEGY)) {
+                    this.handleChildrenLayoutStrategy(jsonObject, value);
+                    break;
+                }
+                this.moveLayoutStrategies(value.getAsJsonObject());
+            } else if (value.isJsonArray()) {
+                for (JsonElement arrayElement : value.getAsJsonArray()) {
+                    if (arrayElement.isJsonObject()) {
+                        this.moveLayoutStrategies(arrayElement.getAsJsonObject());
+                    }
+                }
+            }
+        }
+    }
+
+    private void handleChildrenLayoutStrategy(JsonObject jsonObject, JsonElement value) {
+        jsonObject.remove(CHILDREN_LAYOUT_STRATEGY);
+
+        this.addLayoutStrategyToStyle(jsonObject, value);
+        this.addLayoutStrategyToConditionalStyles(jsonObject, value);
+        this.addLayoutStrategyToChildrenDescriptions(jsonObject);
+    }
+
+    private void addLayoutStrategyToStyle(JsonObject jsonObject, JsonElement value) {
+        JsonObject style = jsonObject.getAsJsonObject(STYLE);
+        if (style != null) {
+            JsonObject styleData = style.getAsJsonObject(DATA);
+            if (styleData != null) {
+                styleData.add(CHILDREN_LAYOUT_STRATEGY, value);
+            }
+        }
+    }
+
+    private void addLayoutStrategyToConditionalStyles(JsonObject jsonObject, JsonElement value) {
+        JsonArray conditionalStyleArray = jsonObject.getAsJsonArray(CONDITIONAL_STYLES);
+        if (conditionalStyleArray != null) {
+            for (JsonElement conditionalStyle : conditionalStyleArray) {
+                JsonObject conditionalStyleData = conditionalStyle.getAsJsonObject().getAsJsonObject(DATA);
```

Comment:
This could probably be a tiny bit more readable with Optional#map which would check for null